### DISCUSSION TITLE
How to handle file extensions with varying casings (lower, upper, etc) before file extension validation is performed when uploading a file

### DISCUSSION THREAD

**Discussion by Benez11** [adonisjs/core#2829] [Help]
Hi Guys!

Would love to be able **_to convert all file extension variations_** (which can be lower - `.exe`, upper - `.JPG`, sentence - `.Png` or even random - `.pDf`) from client **_to lowercase_** so that I can **_compare_** them with a **_lowercase list of allowed file extensions_** and determine if the file possesses an allowed extension at the point the file upload is validated at: ``request.file(<inputName>, { extnames: <allowedExtensions>, size, ... })``

What Happens Now:
- Allowed extensions = `[ 'exe', 'pdf', 'jpg', 'png' ]`
- File comes from client as `Universe.JpG`
- Extension validation method _throws_ `Error => The file extension, '.JpG' is invalid. Allowed Extensions = 'exe', 'pdf', 'jpg', 'png'`

What I Want To Happen:
- Allowed extensions = `[ 'exe', 'pdf', 'jpg', 'png' ]`
- File comes from the client as `Universe.JpG`
- Extension validation method _'passes'_ the file (by converting its extension to lowercase before looking for it in allowed extensions).

What I'd Love To Know:
1. Does this exist already?
2. Can I make this edit and create a PR?
3. What do you guys think about this question?

**Comment by thetutlage** [comment]
At first glance it looks like the extension check should be case in-sensitive. But I would like to see how other frameworks like Rails or Laravel are dealing with this issue

**Reply by arthurcvm** [reply]
@thetutlage we can use a custom middleware to convert to lowercase or create a custom validator equal file validator to validate without case sensitive?

**Reply by Benez11** [reply]
@arthurcvm  I agree. I would love to transform the extension's case to lowercase myself before it is validated (custom middleware).
